packages/webpack/react-webpack-plugin/src/loaders/react-compiler.ts

```diff
@@ -0,0 +1,97 @@
+// Copyright 2025 The Lynx Authors. All rights reserved.
+// Licensed under the Apache License Version 2.0 that can be found in the
+// LICENSE file in the root directory of this source tree.
+import { createRequire } from 'node:module';
+
+import type { PluginItem } from '@babel/core';
+import type { LoaderContext } from '@rspack/core';
+
+import type { ReactLoaderOptions } from './options.js';
+
+const require = createRequire(import.meta.url);
+
+const missingBabelPackages: string[] = [];
+const [
+  swcReactCompilerPath,
+  babelPath,
+  babelPluginReactCompilerPath,
+  babelPluginSyntaxJsxPath,
+  babelPluginSyntaxTypescriptPath,
+] = [
+  '@swc/react-compiler',
+  '@babel/core',
+  'babel-plugin-react-compiler',
+  '@babel/plugin-syntax-jsx',
+  '@babel/plugin-syntax-typescript',
+].map((name) => {
+  try {
+    return require.resolve(name, {
+      paths: [process.cwd()],
```

Comment:
Is is correct to use `process.cwd()`? Maybe it's better to use `loaderContext.rootContext`